app/ui/src/main/java/com/fsck/k9/fragment/MessageListFragment.java

```diff
@@ -2693,8 +2694,12 @@ private void resetActionMode() {
     }
 
     private void startAndPrepareActionMode() {
-        actionMode = getActivity().startActionMode(actionModeCallback);
-        actionMode.invalidate();
+        AppCompatActivity activity = (AppCompatActivity) requireActivity();
+        ActionMode actionMode = activity.startSupportActionMode(actionModeCallback);
```

Comment:
Is there a reason you use `actionMode` local & also assign to the `actionMode` field instead of just using the field (like the previous code)? (performance reasons?)

Reply:
> Also, writing a non-null value to and then immediately reading from a field isn't guaranteed to get you a non-null value, because concurrent modifications are possible.

I would rather expect the opposite - that writing to this field from another thread would *not* be visible as `actionMode` is not volatile and there are no explicit barriers here (no `synchronized`, AtomicReference etc.)

Thanks for the explanation! I think it's good to be merged.